garnett/ciffilereader.py

```diff
@@ -29,20 +30,34 @@
 
 CIFFILE_FLOAT_DIGITS = 11
 
-
 # invalid characters for symmetry expressions
 REMOVE_NONNUM_REGEXP = re.compile(r'[^+\-*/0-9\.,\(\)xyz]+')
 
 # (an integer or floating point number)/(an integer or floating point number)
 PARSE_DIVISION_REGEXP = re.compile(r'(?P<num>\d+(\.(\d+)?)?)\s*/\s*(?P<denom>\d+(\.(\d+)?)?)')
 
+# This regex matches _atom_site_label component 0 (but not component 1) of the
+# definition here:
+# https://www.iucr.org/__data/iucr/cif/standard/cifstd15.html
+_ATOM_SITE_LABEL_COMPONENT_0 = re.compile(r'^(([^ _0-9])|([0-9]*[\+\-]))*')
```

Comment:
Maybe a `\s` instead of `' '` as an extra layer of prevention against tab characters and the like from getting integrated into type names?

Reply:
@klarh I interpreted [this definition](https://www.iucr.org/__data/iucr/cif/standard/cifstd15.html) as a literal space, but I wondered about `\s` as well. How do you read "A character string containing any character except a blank or an underline"?

Reply:
I was interpreting "blank" as "a whitespace character", not the singular "ASCII space". I think it's not totally unambiguous, though.

Reply:
@klarh If you think `\s` is right, commit this suggestion.
```suggestion
_ATOM_SITE_LABEL_COMPONENT_0 = re.compile(r'^(([^\s_0-9])|([0-9]*[\+\-]))*')
```